### ISSUE TITLE
make binary parser error messages spec compliant (see the diff in #251)

### ISSUE THREAD

**Issue by zapashcanon** [OCamlPro/owi#255]

**Comment by zapashcanon** [comment]
Some messages are still wrong (see #262) but it's getting better

**Comment by epatrizio** [comment]
> Some messages are still wrong (see #262) but it's getting better

Indeed. Full compliance seems hard to achieve : our implementation approach is different and the choice of message is sometimes "subjective". But yes, this PR has improved owi :)
